mautic/mautic
# Feature - send scheduled broadcasts command

Please answer the following questions. 

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Related user documentation PR URL | todo
| Related developer documentation PR URL | todo
| Issues addressed (#s or URLs) | #574
| BC breaks? | n
| Deprecations? | n

**Note that all new features should have a related user and/or developer documentation PR in their respective repositories.** 

### Required
#### Description:

This command provides support to schedule broadcasts to be sent and handled via cron. Segment emails (our only channel that supports broadcasts at the moment) can now set a published up/down date and this command will automatically send the emails to any contact pending. 

It also now auto-bounces contacts with poorly formatted email addresses.

#### Steps to test this PR:
1. Create a new segment with several contacts - ensure one has a bad email address
2. Create a new segment email and choose the segment just created - also be sure that the email is publisehd and has a published up date (the command will ignore all emails that do not have a published up date for BC purposes - it would be bad if it back sent all the people pending on very old emails)
3. Run the command from CLI - you should get a table result of the number of successes and failures.
4. Check the contact with the bad email and it should now have a bounce/DNC record.
